cogs/habit_tracker_cog.py

```diff
@@ -0,0 +1,582 @@
+import logging
+import pytz
+import discord
+import datetime
+import aiohttp
+from tabulate import tabulate
+from typing import Tuple
+import config
+import io
+from discord.ext import tasks, commands
+from discord_slash import SlashContext, cog_ext
+from discord_slash.utils.manage_commands import create_option
+from datetime import datetime, timedelta
+from discord import Guild
+
+_FREQUENCY_OPTIONS = ["weekly", "monthly"]
+
+
+class HabitTrackerCog(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.check_habits.start()
+        self.max_habits_per_guild = 2
+
+    def get_period_bounds(self, frequency: str) -> Tuple[datetime, datetime]:
+        """
+        Gets the bounds for the current period depending on the frequency selected
+        """
+        now = datetime.now(pytz.utc)
+
+        if frequency == "weekly":
+            # weekday(): Monday=0 ... Sunday=6
+            # We want to shift so Sunday=0 ... Saturday=6
+            # Calculate days since last Sunday
+            days_since_sunday = (now.weekday() + 1) % 7
+            start = (now - timedelta(days=days_since_sunday)).replace(
+                hour=0, minute=0, second=0, microsecond=0
+            )
+            end = start + timedelta(days=7)
+
+        # Local testing code
+        # elif frequency == "10seconds":
+        #     # Calculate the start of the current 10-second interval
+        #     # e.g. if seconds = 34, start = seconds 30, end = seconds 40
+        #     seconds = now.second
+        #     interval_start_seconds = (seconds // 10) * 10
+        #     start = now.replace(second=interval_start_seconds, microsecond=0)
+        #     end = start + timedelta(seconds=10)

```

Comment:
Instead of commenting out sections for dev, can you please use the `config.STAGE != 'prod'`?